buildNpmPackage: support makeWrapperArgs (#258384)
NixOS/nixpkgs: doc/languages-frameworks/javascript.section.md

}
```

> In the default `installPhase` set by `buildNpmPackage`, it uses `npm pack --json --dry-run` to decide what files to install in `$out/lib/node_modules/$name/`, where `$name` is the `name` string defined in the package's `package.json`. Additionally, the  `bin` and `man` keys in `${src}`'s `package.json`, are used to decide what binaries and manpages are supposed to be installed. If these are not defined, `npm pack` may miss some files, and no binaries will be produced.

Reviewer: ```suggestion
In the default `installPhase` set by `buildNpmPackage`, it uses `npm pack --json --dry-run` to decide what files to install in `$out/lib/node_modules/$name/`, where `$name` is the `name` string defined in the package's `package.json`. Additionally, the  `bin` and `man` keys in the source's `package.json` are used to decide what binaries and manpages are supposed to be installed. If these are not defined, `npm pack` may miss some files, and no binaries will be produced.
```

Maybe don't use `${src}` since if `sourceRoot` or `npmRoot` are set, the actual `package.json` location may be different

* `npmBuildFlags`: Flags to pass to `npm run ${npmBuildScript}`.
* `npmPackFlags`: Flags to pass to `npm pack`.
* `npmPruneFlags`: Flags to pass to `npm prune`. Defaults to the value of `npmInstallFlags`.

> * `makeWrapperArgs`: Flags to pass to `makeWrapper`, added to the node wrapper calling the generated `.js` scripts.

Reviewer: Maybe make it clearer this is only used for `bin`s defined in package.json?

```suggestion
* `makeWrapperArgs`: Flags to pass to `makeWrapper` when wrapping package-defined `bin` entrypoints with `node`.
```

I'm open to wording suggestions, since it's a bit early this morning for me to write scrutable documentation, but this is just a suggestion of what I'm thinking

* `npmBuildFlags`: Flags to pass to `npm run ${npmBuildScript}`.
* `npmPackFlags`: Flags to pass to `npm pack`.
* `npmPruneFlags`: Flags to pass to `npm prune`. Defaults to the value of `npmInstallFlags`.

> * `makeWrapperArgs`: Flags to pass to `makeWrapper`, added to executable calling the generated `.js` with `node` as an interpreter. These scripts are defined in `

Reviewer: ```suggestion
* `makeWrapperArgs`: Flags to pass to `makeWrapper`, added to executable calling the generated `.js` with `node` as an interpreter. These scripts are defined in `package.json`.
```

Typo?

sha256-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=
```

#### fetchNpmDeps {#javascript-buildNpmPackage-fetchNpmDeps}

`fetchNpmDeps` is a Nix function that requires the following mandatory arguments:

- `src`: A directory / tarball with `package-lock.json` file
- `hash`: The output hash of the node dependencies defined in `package-lock.json`.

> And it returns a derivation with all `package-lock.json` dependencies downloaded into `$out/`, useable as a NodeJS cache, when you need to access it manually.

Reviewer: ```suggestion
It returns a derivation with all `package-lock.json` dependencies downloaded into `$out/`, usable as an npm cache.
```

Tiny voice nit (feel free to ignore that one, I don't feel that strongly) and clarifying it's an npm cache, not a generalized Node.js cache
